src/layout/NavBar/components/NavBar.jsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import MenuIcon from '@material-ui/icons/Menu';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+
+import { ModalView } from 'components/';
+import AuthContainer from 'layout/Auth';
+
+import { withStyles } from '@material-ui/core/styles';
+
+import { AuthStates } from 'redux/actions.js';
+
+const { GUEST } = AuthStates;
+
+// If the user is a guest, show a Sign In Button
+// Use the navigate and logout props
+
+// TODO: Render the auth view in a modal instead
+export const MenuButton = (props) => {
+  const { toggle } = props;
+  return (
+    <IconButton onClick={toggle} color='inherit'>
+      <AccountCircle />
+    </IconButton>
+  );
+};
+
+MenuButton.propTypes = {
+  toggle: PropTypes.func.isRequired,
+};
+
+// Opens up the Auth Modal when clicked
+export const SignUpButton = (props) => {
+  const { toggleAuth } = props;
+  return (
+    <Button onClick={toggleAuth} color='inherit'>
+      Sign Up
+    </Button>
+  );
+};
+
+SignUpButton.propTypes = {
+  toggleAuth: PropTypes.func.isRequired,
+};
+
+const styles = {
```

Comment:
We should consider putting styles in a different file. It's not a big deal here, but I've seen style objects get really big. What do you think?

Reply:
Well i mean I really don't want to make a whole other file for this one in particular. Most of the styling is done by the component library so to make a new file for maybe 5 lines of uncondensed style might just be clutter. For the Content and Body of the pages, we should make a styles object in a different file.

Reply:
Sounds good